dataset_interfaces/interface.py

```diff
@@ -193,6 +196,52 @@ def load(cls, file_path: Path | str) -> "TestExample":
         d = {k: d[k] for k in d.keys() if k not in generator_attrs}
         return dataset_generator.create_example(**d)
 
+    def default_waits(self):
+
+        if isinstance(self, DynamicExample):
+            return
+
+        assert len(self.is_question) >= 1, "There are no questions for this test"
+        assert len(self.waits) == 0 or len(self.waits) == len(self.is_question), "Current waits should be empty or the same length as the script"
+
+        # TODO: All this token counting is based on OpenAIs methods, which is not Anthropics - but Aanthopic has no reliable tokeniser released so....
+        script_tokens = get_tokens_for_script(self.script)
+        assert script_tokens < self.memory_span, f"The script for test {self.dataset_name} is too long (estimated {script_tokens} tokens) for the specified memory span of {self.memory_span} tokens."
+
+        span_minus_script = self.memory_span - script_tokens
+        needle_question_proportions = (.75, .25) if self.number_of_questions == 1 else (.5, .5)
+
+        # Figure out what our spacing is going to be. We will do the 75%/25% needle distribution.
+        tokens_for_needles = math.floor(span_minus_script * needle_question_proportions[0])
+        needles = len(self.is_question) - sum(self.is_question)
```

Comment:
This would need setting some constraints in relation to needle placement e.g. needles go first and after that there can only be questions. This is untrue for the restaurant task, which constantly interleaves needles and questions.

Reply:
Yeah, this function is pretty much incompatible with the restaurant task, and probably dynamic tasks in general